src/somd2/runner/_runner.py

```diff
@@ -185,19 +189,129 @@ def _check_space(self):
 
     def _check_directory(self):
         """
-        Check if the output directory has files already present.
-        Paired with the 'overwrite' option.
+        Find the name of the file from which simulations will be started.
+        Paired with the 'restart' option.
         """
-        import os as _os
-        import glob as _glob
+        from pathlib import Path as __Path
+        from ._dynamics import Dynamics
 
-        if self._config.overwrite:
-            return True
-        else:
-            if _glob.glob(_os.path.join(self._config.output_directory, "*.parquet")):
-                return False
+        fnames = {}
+        deleted = []
+        for lambda_value in self._lambda_values:
+            files = Dynamics.create_filenames(
+                self._lambda_values,
+                lambda_value,
+                self._config.output_directory,
+                self._config.restart,
+            )
+            fnames[lambda_value] = files
+            if not self._config.restart:
+                for file in files.values():
+                    fullpath = self._config.output_directory / file
+                    if __Path.exists(fullpath):
+                        deleted.append(fullpath)
+        if len(deleted) > 0:
+            if not self._config.supress_overwrite_warning:
+                _logger.warning(
+                    f"The following files already exist and will be overwritten: {list(set((deleted)))} \n"
+                )
+                input("Press Enter to erase and continue...")
+            # Loop over files to be deleted, ignoring duplicates
+            for file in list(set(deleted)):
+                file.unlink()
+        self._fnames = fnames
+
+    @staticmethod
+    def _compare_configs(config1, config2):
+        if not isinstance(config1, dict):
+            raise TypeError("'config1' must be of type 'dict'")
+        if not isinstance(config2, dict):
+            raise TypeError("'config2' must be of type 'dict'")
+
+        # Define the subset of settings that are allowed to change after restart
+        allowed_diffs = [
+            "runtime",
+            "restart",
+            "temperature",
```

Comment:
Do we want to allow the temperature to change between runs? This is baked into the energy trajectory data frame in the `attrs`, which is required by `alchemlyb`. As such, I don't think we can change the temperature mid simulation.

It might also be worth checking how `alchemlyb` deals with unevenly spaced sampling, e.g. what happens if the energy frequency differs for part of the run? The time is a column of the data frame so it _should_ be able to handle this, but I've not checked.